### PR TITLE
Make pagination dynamic, modify function names, get no of users from AbstractRepo function

### PR DESCRIPTION
**Pull request by andreiuta97**
Make pagination dynamic in Services (get rid of the hard-coded "results per page").
Modify function names in Services as Victor recommended at search-quiz. 
Get number of users from AbstractRepository function - getNumberOfObjects.

### CODE REVIEW

File: src/Controller/QuestionTemplateController.php [hunk 1]
```diff
@@ -83,12 +83,12 @@ public function deleteQuestion(Request $request, array $requestAttributes): Resp
     public function getQuestions(Request $request, array $requestAttributes): Response
     {
         $filters = isset($requestAttributes['text']) ? ['text' => $requestAttributes['text']] : [];
-        $count = $this->questionTemplateService->getQuestionNumber($filters);
+        $count = $this->questionTemplateService->getFilteredQuestionsNumber($filters);
```
Comment (Raresmldvn) [inline src/Controller/QuestionTemplateController.php]: I personally would like the method to be called without the "filtered" part, but this is just my option

File: src/Service/QuestionTemplateService.php [hunk 2]
```diff
@@ -52,13 +52,15 @@ public function getQuestion(int $id): QuestionTemplate
         return $question;
     }
 
-    public function getQuestions(array $filters,int $currentPage): array
+    public function getFilteredQuestionsForPage(array $filters,int $currentPage): array
     {
-        $criteria = new Criteria($filters, [], ($currentPage - 1) * 5, 5);
+        $paginator = new Paginator($this->getFilteredQuestionsNumber($filters));
+        $criteria = new Criteria($filters, [], ($currentPage - 1) * $paginator->getResultsPerPage(), $paginator->getResultsPerPage());
```
Comment (Raresmldvn) [inline src/Service/QuestionTemplateService.php]: maybe you should enable the paginator to compute limit and offset, you're using this formula in a lot of places.

File: src/Controller/QuestionTemplateController.php (hunk 1 above)
Comment (victortodoran) [reply src/Controller/QuestionTemplateController.php]: @Raresmldvn @andreiuta97 I have no attachments on the naming I suggested if you consider it best to use another one, please do.

Comment (Raresmldvn) [reply src/Controller/QuestionTemplateController.php]: no, it's ok :D I said it's just a personal preference, but I'm good with this

File: src/Controller/QuestionTemplateController.php [hunk 3]
```diff
@@ -80,18 +82,24 @@ public function deleteQuestion(Request $request, array $requestAttributes): Resp
...
+    private function getCriteriaFromRequest(array $requestAttributes): Criteria
     {
         $filters = isset($requestAttributes['text']) ? ['text' => $requestAttributes['text']] : [];
-        $count = $this->questionTemplateService->getQuestionNumber($filters);
-        $paginator = new Paginator($count);
-        if (isset($requestAttributes['page'])) {
-            $paginator->setCurrentPage($requestAttributes['page']);
-        }
-        $questions = $this->questionTemplateService->getQuestions($filters, $paginator->getCurrentPage());
+        $currentPage = isset($requestAttributes['page']) ? $requestAttributes['page'] : 1;
```
Comment (Raresmldvn) [inline src/Controller/QuestionTemplateController.php]: can be written as `$currentPage = $requestAttributes['page'] ?? 1;` :D just saying

Comment (victortodoran) [reply src/Controller/QuestionTemplateController.php]: @Raresmldvn if $requestAttributes does not have value for 'page' wouldn't it result in a PHP error?

Comment (Raresmldvn) [reply src/Controller/QuestionTemplateController.php]: no, just checked. it integrates with isset

Comment (victortodoran) [reply src/Controller/QuestionTemplateController.php]: @Raresmldvn cool

Comment (Raresmldvn) [reply src/Controller/QuestionTemplateController.php]: @andreiuta97 if you change this, merge after, if not, you can merge now if Victor also agrees

Comment (andreiuta97) [reply src/Controller/QuestionTemplateController.php]: @Raresmldvn, I'm gonna change it first.

**Pull request merged**